BaseTools/Edk2ToolsBuild.py

```diff
@@ -110,27 +115,122 @@ def Go(self):
         shell_env.set_shell_var("PYTHON_COMMAND", pc)
 
         if self.tool_chain_tag.lower().startswith("vs"):
+            if self.target_arch is None:
+                # Put a default as IA32
+                self.target_arch = "IA32"
+
+            if self.target_arch == "IA32":
+                VcToolChainArch = "x86"
+                TargetInfoArch = "x86"
+                OutputDir = "Win32"
+            elif self.target_arch == "ARM":
+                VcToolChainArch = "x86_arm"
+                TargetInfoArch = "ARM"
+                OutputDir = "Win32"
+            elif self.target_arch == "X64":
+                VcToolChainArch = "amd64"
+                TargetInfoArch = "x86"
+                OutputDir = "Win64"
+            elif self.target_arch == "AARCH64":
+                VcToolChainArch = "amd64_arm64"
+                TargetInfoArch = "ARM"
+                OutputDir = "Win64"
+            else:
+                raise NotImplementedError()
 
-            # # Update environment with required VC vars.

```

Comment:
This unrelated change to the comment text confuses the diff somewhat.
Please put back the redundant #.

Reply:
This was reverted. It indeed looks better on git diff.